cassandra/src/main/scala/filodb.cassandra/metastore/IngestionStateTable.scala

```diff
@@ -0,0 +1,100 @@
+                                (implicit val ec: ExecutionContext) extends FiloCassandraConnector {
+
+  val keyspace = config.getString("admin-keyspace")
+  val tableString = s"${keyspace}.ingestion_state"
+
+  val createCql = s"""CREATE TABLE IF NOT EXISTS $tableString (
+                      |nodeactor text,
+                      |database text,
+                      |dataset text,
+                      |version int,
+                      |columns text,
+                      |state text,
+                      |exceptions text,
+                      |PRIMARY KEY (nodeactor, database, dataset, version)
+                      |)""".stripMargin
+
+  import filodb.cassandra.Util._
+
+  lazy val ingestionStateInsertCql = session.prepare(
+    s"""INSERT INTO $tableString (nodeactor, database, dataset, version, columns, state)
+       | VALUES (?, ?, ?, ?,?, ?) IF NOT EXISTS""".stripMargin
+  )
+
+  lazy val ingestionStSelectByDsCql = session.prepare(
+    s"""SELECT nodeActor, database, dataset, version, columns, state, exceptions
+       | FROM $tableString WHERE nodeactor = ? AND database = ? AND dataset = ? AND version = ?""".stripMargin
+  )
+
+  lazy val ingestionStSelectByActorCql = session.prepare(
+    s"""SELECT nodeActor, database, dataset, version, columns, state, exceptions
+       | FROM $tableString WHERE nodeactor = ? """.stripMargin
+  )
+
+  def initialize(): Future[Response] = execCql(createCql)
+
+  def fromRow(row: Row): IngestionStateData =
+    IngestionStateData(row.getString("nodeActor"),
+      row.getString("database"),
+      row.getString("dataset"),
+      row.getInt("version"),
+      row.getString("columns"),
+      row.getString("state"),
+      row.getString("exceptions"))
+
+  def clearAll(): Future[Response] = execCql(s"TRUNCATE $tableString")
+
+  def dropTable(): Future[Response] = execCql(s"DROP TABLE $tableString")
+
+  def insertIngestionState(nodeActor: String, database: String, name: String,
+                           version: Int, columns: String, state: String): Future[Response] = {
+    execStmt(ingestionStateInsertCql.bind(
+              nodeActor, database, name, version: java.lang.Integer, columns, state),
+              AlreadyExists
+            )
+  }
+
+  def getIngestionStateByDataset(nodeActor: String, database: String,
+                        name: String, version: Int) : Future[Seq[Row]] =
+    session.executeAsync(ingestionStSelectByDsCql.bind(
+                              nodeActor,
+                              database,
+                              name,
+                              version: java.lang.Integer)
+                        ).toIterator.map { it => it.toSeq }
+
+  def getIngestionStateByNodeActor(nodeActor: String) : Future[Seq[IngestionStateData]] =
+    session.executeAsync(ingestionStSelectByActorCql.bind(
+                            nodeActor)
+                        ).toIterator.map {_.map(fromRow).toSeq}
```

Comment:
you mean remove toIterator before map or just put in other line. Getting compile error if i remove toIterator before map.